doc/model.md

```diff
@@ -0,0 +1,74 @@
+# Istio Service Model
+
+Istio's services model introduces the concept of a service version, a
+finer-grained service notion used to subdivide service instances by
+versions (`v1`, `v2`) or environment (`staging`, `prod`). Istio routing
+rules can refer to the service versions, to provide additional control over
+traffic between services.
+
+The Istio model of services in the application is independent of the
+underlying platform (Kubernetes, Mesos, CloudFoundry, etc.). Platform
+specific adapters are responsible for populating the internal model
+representation with various fields, from the metadata found in the
+platform. The following sections describe various aspects of the abstract
+service model used by Istio.
+
+## Service
+
+Service is a unit of an application with a unique name that other services
+use to refer to the functionality being called. Each service is referred to
+using a fully qualified domain name (FQDN) and one or more ports where the
+service is listening for connections. Each service has one or more
+instances, i.e., actual manifestations of the service.  Instances represent
+entities such as containers, pods (kubernetes/mesos), VMs, etc.
```

Comment:
We probably want to add some text here to describe the notion of a tenant, and of a multi-tenant service, as it relates to service instances in particular.